### PR TITLE
New configuration docs. (#73)

### PR DESCRIPTION
**Pull request by eyalb181**

### CODE REVIEW

File: content/docs/overview/configuration.md [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+
+IP:PORT to connect to instead of using k8s api, for testing purposes.
+
+```json
+{
+  "connect_tcp": "10.10.0.100:7777"
+}
+```
+
+## connect_agent_name {#root-connect_agent_name}
```
Comment (eyalb181) [inline content/docs/overview/configuration.md:194]: Is this ever supposed to be configured by the user?

File: content/docs/overview/configuration.md [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+aren't stolen (due to being already established)
+
+Defaults to `true`.
+
+# feature {#root-feature}
+
+Controls mirrord features.
+
+See the
+[technical reference, Technical Reference](https://mirrord.dev/docs/reference/)
```
Comment (eyalb181) [inline content/docs/overview/configuration.md:370]: ?

**Review by eyalb181 — COMMENTED** [review]
Whoops I'm the PR author! Anyway made some textual change requests. Otherwise looks really good!

**Review by eyalb181 — COMMENTED** [review]
Whoops I'm the PR author! Anyway made some textual change requests. Otherwise looks really good!

File: content/docs/overview/configuration.md (hunk 1 above)
Comment (meowjesty) [reply content/docs/overview/configuration.md:194]: That's a good question I don't have an answer for.

We have comments on a couple of features that are internal, but there were no comments on this one.

File: content/docs/overview/configuration.md [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+menu:
+  docs:
+    parent: "overview"
+weight: 121
+toc: true
+---
+
+# Setting up mirrord {#root}
+
+Mirrord allows for a high degree of customization when it comes to which features you want to
```
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md]: All lowercase now.

File: content/docs/overview/configuration.md [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+enable, and how they should function.
+
+Mirrord features can be setup with the [`feature`](#root-feature) option.
```
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md]: Alright, removed.

File: content/docs/overview/configuration.md [hunk 5]
```diff
@@ -0,0 +1,783 @@
...
+
+## target {#root-target}
+
+Specifies the target and namespace to mirror, see [`path`](#target-path) for a list of accepted
+values for the `target` option.
+
+The simplified configuration supports:
+
+- `pod/{sample-pod}/[container]/{sample-container}`;
+- `podname/{sample-pod}/[container]/{sample-container}`;
```
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md:140]: Don't recall specifics, just remember that these are all the things we support.

File: content/docs/overview/configuration.md [hunk 6]
```diff
@@ -0,0 +1,783 @@
...
+  "connect_tcp": "10.10.0.100:7777"
+}
+```
+
+## connect_agent_name {#root-connect_agent_name}
+
+Agent name that already exists that we can connect to.
+
+Keep in mind that the intention here is to allow reusing a long living mirrord-agent pod, and 
+**not** to connect multiple (simultaneos) mirrord instances to a single mirrord-agent, as the later
```
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md]: Oh I was talking about the `SimultaneOS`, this is not a typo!

...

Fixed

File: content/docs/overview/configuration.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md]: > no difference

Removed the `podname` and `containername`.

File: content/docs/overview/configuration.md (hunk 1 above)
Comment (meowjesty) [reply content/docs/overview/configuration.md]: > yes, strip!

Instructions unclear, taking off my clothes.

File: content/docs/overview/configuration.md [hunk 7]
```diff
@@ -0,0 +1,762 @@
...
+```
+
+## target {#root-target}
+
+Specifies the target and namespace to mirror, see [`path`](#target-path) for a list of accepted
+values for the `target` option.
+
+The basic configuration supports:
+
+- `pod/{sample-pod}/[container]/{sample-container}`;
```
Comment (eyalb181) [inline content/docs/overview/configuration.md:139]: I think these examples are confusing and we should instead give the examples in line 171

File: content/docs/overview/configuration.md [hunk 8]
```diff
@@ -0,0 +1,777 @@
...
+draft: false
+images: []
+menu:
+  docs:
+    parent: "overview"
+weight: 121
+toc: true
+---
+
+# Setting up mirrord {#root}
```
[the remark this answers is not in the document]
Comment (meowjesty) [reply content/docs/overview/configuration.md]: On the topic of headers (somewhat).

Should I sort the options in a way that root config comes first, then we have the options that have sub-configs?

Kinda like

1. skip_processes
2. operator
3. ...
4. target
5. agent
6. feature

[the remark this answers is not in the document]
Comment (eyalb181) [reply content/docs/overview/configuration.md]: Sure, makes sense

**Review by meowjesty — APPROVED** [review]
Thank you for this PR @eyalb181 will greatly help our users!

I should send you something nice, like a cake, for such an amazing PR.

**Pull request merged**